syntax = "proto3";

package Grpc;

//Data Base Service
service DataBaseService {
		
	rpc AddUser(User) returns (AddUserResponse) {}	
	
	rpc GetUserByLogin(Login) returns (GetUserResponse) {}

	rpc GetUserByMail(Mail) returns (GetUserResponse) {}
	
	rpc GetAllLogins(VoidRequest) returns (stream Login) {}
	
	rpc GetPasswordByLogin(Login) returns (Password) {}
		
	rpc IsExist(Login) returns (IsExistResponse) {}
}

message VoidRequest {
}

message AddUserResponse {

	string error_message = 1;
}

message JsonDocument {
	string json = 1;
}

message Login {
	string login = 1;
}

message Password {
	string password = 1;
}

message Mail {
	string mail = 1;
}

message User {
	Login login = 1;
	Password password = 2;
	Mail mail = 3;
}

message GetUserResponse {
	User user = 1;
	string error_message = 2;
	JsonDocument json_user = 3;
}

message IsExistResponse {
	bool exist = 1;
}



//Chat Service
service ChatService {
		
	rpc SendMessage(SendMessageRequest) returns (IsOkResponse) {}
	
	rpc GetFullChat(PrivateChatUsers) returns (FullChat) {}

}

message IsOkResponse {
	bool ok = 1;
}

message SendMessageRequest {
	PrivateChatUsers users = 1;
	Message text = 2;
}

message PrivateChatUsers {
	User sender = 1;
	User reciver = 2;
}

message Date {
	int32 year = 1;	
	int32 month = 2;
	int32 day = 3;
	int32 hours = 4;
	int32 minutes = 5;
}

message Message {
	string author_login = 1;
	string value = 2;
	Date send_date = 3;
}

message FullChat {
	repeated Message messages= 2;
}






